05_stringer/starfallz/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, IPAddr{127, 0, 0, 1})
+}
+
+type IPAddr struct {
+	add1, add2, add3, add4 uint8
+}
+
+func (ipAddr IPAddr) String() string {
+	return strings.Join([]string{
```

Comment:
You can use a string format instead of joining on a slice
`fmt.Sprintf("%d.%d.%d.%d", ipaddr.add1, ipaddr.add2, ...)`

Reply:
dpes fmt.Sprintf return a string not print out the string?

Reply:
https://godoc.org/fmt#Sprintf (it returns a string)

Reply:
https://godoc.org/fmt#Sprintf (it does)